src/witan/models/dem/ccm/components_functions.clj

```diff
@@ -16,12 +18,20 @@
   for all the years of interest and is named to avg-name."
   [historical-data col-to-avg avg-name start-yr-avg end-yr-avg]
   (let [hist-earliest-yr (reduce min (ds/column historical-data :year))
-        hist-latest-yr (reduce max (ds/column historical-data :year))
-        start-yr (s/validate (s/pred #(>= % hist-earliest-yr)) start-yr-avg)
-        end-yr (s/validate (s/pred #(<= % hist-latest-yr)) end-yr-avg)]
+        hist-latest-yr (reduce max (ds/column historical-data :year))]
+    (-> start-yr-avg
+        (utils/property-holds?  #(>= % hist-earliest-yr)
```

Comment:
It isn't much of a difference, but I wonder if we shouldn't first check if it's a year before checking it's in the right range.